src/workerd/api/filesystem.c++

```diff
@@ -59,6 +59,22 @@ struct NormalizedFilePath {
   void validate() {
     JSG_REQUIRE(url.getProtocol() == "file:"_kj, TypeError, "File path must be a file: URL");
     JSG_REQUIRE(url.getHost().size() == 0, Error, "File path must not have a host");
+
+    // Let's count the number of path segments in the URL. We want to make sure that
```

Comment:
We should document this on the docs pull request.

Reply:
Yep. all the limits will be documented.